test/e2e/upgrade_ironic_test.go

```diff
@@ -0,0 +1,73 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func upgradeIronic() {
+	var (
+		namePrefix        = e2eConfig.GetVariable("NAMEPREFIX")
+		clientSet         = targetCluster.GetClientSet()
+		ironicNamespace   = e2eConfig.GetVariable("IRONIC_NAMESPACE")
+		ironicDeployName  = namePrefix + "-ironic"
+		containerRegistry = e2eConfig.GetVariable("CONTAINER_REGISTRY")
+		ironicImageTag    = e2eConfig.GetVariable("IRONIC_IMAGE_TAG")
+	)
+
+	Logf("namePrefix %v", namePrefix)
+	Logf("ironicNamespace %v", ironicNamespace)
+	Logf("ironicDeployName %v", ironicDeployName)
+	Logf("containerRegistry %v", containerRegistry)
+	Logf("ironicImageTag %v", ironicImageTag)
+
+	By("Upgrading ironic image based containers")
+	deploy, err := getIronicDeployment(clientSet, ironicDeployName, ironicNamespace)
+	Expect(err).To(BeNil())
+	for i, container := range deploy.Spec.Template.Spec.Containers {
+		switch container.Name {
+		case
+			"mariadb",
+			"ironic-api",
+			"ironic-dnsmasq",
+			"ironic-conductor",
+			"ironic-log-watch",
+			"ironic-inspector":
+			deploy.Spec.Template.Spec.Containers[i].Image = containerRegistry + "/metal3-io/ironic:" + ironicImageTag
```

Comment:
Tiny improvement: you could swap `i` for `_` and use `container.Image` here.

Reply:
I have tried this and add a description to the previous comment above but it did not work since that way `deploy` object will not be updated.

Reply:
Oops, yes, sorry. This would have to be a slice to `*Container` for this to work. And it's not. Indexing is totally fine.